### ISSUE TITLE
Passwords not retrievable on OS X 10.11.6 / Chrome 62.0.3202.94

### ISSUE THREAD

**Issue by NateEag** [hsanson/chrome-pass#13]
Clicking the extension's icon lists my stored passwords and I can search them, so much of this is working, but when I click on a password to retrieve it, I see the message "No response from native app."

I tried the diagnostics at https://github.com/hsanson/chrome-pass/issues/8 and verified that `nativePass` can indeed decrypt my passwords.

Any clues how I might debug this further?

**Comment by hsanson** [comment]
I have the same chrome version and works perfectly on my PC. If the chrome extension cannot communicate with the native app it could mean that the extension id or the native app path is not set correctly.

Did you follow the troubleshooting section in the README?

**Comment by NateEag** [comment]
Hey, sorry for the slow response - been pretty busy.

I have gone through the troubleshooting section in the readme.

The only thing that's not *exactly* as described is that my native host application manifest is in a different folder, presumably due to being on OS X.

It's in `~/Library/Application\ Support/Google/Chrome/NativeMessagingHosts/com.piaotech.chrome.extension.pass.json`.

Copying it to `~/.config/google-chrome/NativeMessagingHosts` per the troubleshooting section has no effect.

I'll reiterate that I can get the list of usernames fine - it's just when I click one that it fails.

I could probably figure this out myself if I could find the error messages from the nativePass program. Where should I look for those?

**Comment by NateEag** [comment]
aha - I just discovered I can look at the "Inspect views" link on the `chrome://extensions` page.

With that I probably have what I need to figure this out.

**Comment by NateEag** [comment]
If I start Chrome from the command line, per [Google's advice](https://developer.chrome.com/apps/nativeMessaging#native-messaging-debugging) for debugging native messaging, the plugin works.

When I first discovered that, I thought it must have been my .bashrc setting the PATH or similar, but it still worked even when I moved it to ~/.bashrc.bak, started a new terminal, and tried it again.

When I start Chrome through Spotlight, regardless of my .bashrc's location, retrieving passwords still fails.

Judging from [this page on logging](https://www.chromium.org/for-testers/enable-logging), I don't see a way to enable logging without running from the command line, so I'm unsure how to see the error messages for whatever's going wrong.

**Comment by NateEag** [comment]
Chrome started from Terminal.app included /usr/local/bin in its PATH env var, while Chrome started by normal means did not.

I needed it there for Python to find gpg, since I installed gpg via Homebrew.

Customizing environment variables for .app bundles on OS X turns out to be much more fraught with difficulty than I would have guessed.

I finally found [this solution](https://apple.stackexchange.com/a/243946) that did the job for me, and retrieving passwords is working now.

**Comment by NateEag** [comment]
Closed the issue since it's a configuration problem.

It might be worth updating the readme to warn people about this, similar to the Linux "wrong GPG version" issue.

**Comment by hsanson** [comment]
Thanks for the debugging... I added a small comment to the README regarding this possible issue.

**Comment by NateEag** [comment]
You're welcome. Thanks for updating the readme.

**Comment by NateEag** [comment]
A caveat to the solution I applied: 

It does not work in Chrome instances that were auto-started by OS X after a reboot.

At least that appears to be the pattern, and I believe I saw someone else mention an analogous issue back when I was researching this.

So far, when I've hit an instance of the "No response from native app" issue again, quitting Chrome and restarting it has made it go away.

Just wanted to mention that for anyone else who runs into this problem.
